pkg/stanza/fileconsumer/file.go

```diff
@@ -109,164 +132,189 @@ func (m *Manager) startPoller(ctx context.Context) {
 func (m *Manager) poll(ctx context.Context) {
 	// Increment the generation on all known readers
 	// This is done here because the next generation is about to start
+	m.knownFilesLock.Lock()
 	for i := 0; i < len(m.knownFiles); i++ {
 		m.knownFiles[i].generation++
 	}
-
-	// Used to keep track of the number of batches processed in this poll cycle
-	batchesProcessed := 0
+	m.knownFilesLock.Unlock()
 
 	// Get the list of paths on disk
 	matches := m.finder.FindFiles()
-	for len(matches) > m.maxBatchFiles {
-		m.consume(ctx, matches[:m.maxBatchFiles])
+	m.consume(ctx, matches)
+	m.clearCurrentFiles()
 
-		// If a maxBatches is set, check if we have hit the limit
-		if m.maxBatches != 0 {
-			batchesProcessed++
-			if batchesProcessed >= m.maxBatches {
-				return
+	// Any new files that appear should be consumed entirely
+	m.readerFactory.fromBeginning = true
```

Comment:
Commenting here to remind myself that when `start_at: end`, we can only do this if we have at least noted the offset for each file. This will become important if we try to bail out early.

Reply:
Noted